contrib/devtools/test_deterministic_coverage.sh

```diff
@@ -12,6 +12,9 @@ export LC_ALL=C
 # Use GCOV_EXECUTABLE="llvm-cov gcov" if compiling with clang.
 GCOV_EXECUTABLE="gcov"
 
+# Keep random seed constant over multiple runs.
+export RANDOM_CTX_SEED=${RANDOM_CTX_SEED:-8c9678d7cde616175258686f6aaf7c2446657f40a9bc5196ba97ca21b5e3c5c4}
```

Comment:
I think the `contrib/devtools/test_deterministic_coverage.sh` script will fail to detect non-deterministic coverage if the seed is fixed between runs. Isn't it the whole point of that script?
To make sure I ran a single test, `wallet_tests/coin_mark_dirty_immature_credit` 30 times using the script and I always got the same coverage.
If I'm not mistaken, the main purpose here is to log the seed, so maybe this should be changed to a random number read from /dev/random, something like
`export RANDOM_CTX_SEED=$(xxd -l 32 -p /dev/urandom | tr -d " \n")`
Also, for this purpose, I believe this should be moved to within the `while [[ ${TEST_RUN_ID} -lt ${N_TEST_RUNS} ]]; do` loop so that the seed will be different for each iteration.

Reply:
Thanks, switched to `xxd -l 32 -p /dev/urandom | tr -d " \n"`